test/SparqlAntlrParserTest.cpp

```diff
@@ -874,28 +873,22 @@ TEST(SparqlParser, SelectQuery) {
       m::GraphPattern(m::Triples({{"?x", "?y", "?z"}}));
   expectSelectQuery(
       "SELECT * WHERE { ?a <bar> ?foo }",
-      testing::AllOf(
-          m::SelectQuery(m::AsteriskSelect(), m::GraphPattern(m::Triples(
-                                                  {{"?a", "<bar>", "?foo"}}))),
-          m::pq::OriginalString("SELECT * WHERE { ?a <bar> ?foo }")));
-  expectSelectQuery(
-      "SELECT * WHERE { ?x ?y ?z }",
-      testing::AllOf(
-          m::SelectQuery(m::AsteriskSelect(), DummyGraphPatternMatcher),
-          m::pq::OriginalString("SELECT * WHERE { ?x ?y ?z }")));
-  expectSelectQuery(
-      "SELECT ?x WHERE { ?x ?y ?z } GROUP BY ?x",
-      testing::AllOf(
-          m::SelectQuery(m::VariablesSelect({"?x"}), DummyGraphPatternMatcher),
-          m::pq::OriginalString("SELECT ?x WHERE { ?x ?y ?z } GROUP BY ?x"),
-          m::pq::GroupKeys({Variable{"?x"}})));
+      testing::AllOf(m::SelectQuery(
+          m::AsteriskSelect(),
+          m::GraphPattern(m::Triples({{"?a", "<bar>", "?foo"}})))));
+  expectSelectQuery("SELECT * WHERE { ?x ?y ?z }",
+                    testing::AllOf(m::SelectQuery(m::AsteriskSelect(),
+                                                  DummyGraphPatternMatcher)));
+  expectSelectQuery("SELECT ?x WHERE { ?x ?y ?z } GROUP BY ?x",
+                    testing::AllOf(m::SelectQuery(m::VariablesSelect({"?x"}),
+                                                  DummyGraphPatternMatcher),
+                                   m::pq::GroupKeys({Variable{"?x"}})));
```

Comment:
If I understand this correctly, you have dropped the `OriginalString` test, because these strings are always the same. But can you still test this (maybe in the `expectSelectQuery` because the original string has to be equal. This seems trivial now, but makes the tests more robust towards future unintended changes.

Reply:
No I dropped it because the original string now is not set in `visitTypesafe(SelectQuery)` but in the visit for `Query`. So it is only set if something is parsed as a `Query`. If something is parsed as `SelectQuery` it is not set.